### PR TITLE
Feature/scrum 7/add to the catlog

### PR DESCRIPTION
**Pull request by LiangbingQiangweiKaishaHexi1314**
References
Description:
Change Type (feat, fix, etc.):

Code Review Checklist:

- [x] Tested feature manually via emulator
- [x] Created and passed all test cases
- [x] Consistent formatting
- [x] Clean code with no errors or warnings
- [x] Adheres to SOLID principles

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+    <string name="app_name">B07 Demo Summer 2024</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
+    <string name="delete_item">Delete Item</string>
+    <string name="add_item">Add Item</string>
+    <string name="back">Back</string>
+    <string name="recyclerview">RecyclerView</string>
+    <string name="scroller">Scroller</string>
+    <string name="spinner">Spinner</string>
+    <string name="manage_items">Manage Items</string>
```
Comment (David-Huynh) [inline app/src/main/res/values/strings.xml]: Some of this is old can you remove the ones you aren't using?

File: app/build.gradle.kts [hunk 2]
```diff
@@ -37,7 +40,6 @@ android {
             println("All tests executed successfully.")
         }
     }
-    testOptions.unitTests.isIncludeAndroidResources = true
```
Comment (David-Huynh) [inline app/build.gradle.kts]: Not sure why you removed this?

File: app/build.gradle.kts [hunk 3]
```diff
@@ -81,5 +83,16 @@ dependencies {
     // For local unit tests
     testImplementation(libs.dagger.hilt.android.testing)
     testAnnotationProcessor(libs.com.google.dagger.hilt.compiler2)
+
+    implementation(platform("com.google.firebase:firebase-bom:33.1.2"))
+    implementation("com.google.firebase:firebase-analytics")
```
Comment (David-Huynh) [inline app/build.gradle.kts]: Im not sure if we need these two

File: app/build.gradle.kts [hunk 4]
```diff
@@ -81,5 +83,16 @@ dependencies {
...
+
+    // Firebase Storage dependency
+    implementation ("com.google.firebase:firebase-storage:20.0.1")
+    implementation ("com.google.firebase:firebase-database:20.0.3")
+
+    implementation ("com.google.code.gson:gson:2.8.9")
+    implementation ("com.github.bumptech.glide:glide:4.12.0")
+    annotationProcessor ("com.github.bumptech.glide:compiler:4.12.0")
```
Comment (David-Huynh) [inline app/build.gradle.kts:94]: did you end up using these?

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -18,13 +18,14 @@
...
+        android:layout_width="400dp"
+        android:layout_height="700dp"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintHorizontal_bias="0.454"
+        app:layout_constraintHorizontal_bias="1.0"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/fragmentContainerMenu">
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="1.0">
```
Comment (David-Huynh) [inline app/src/main/res/layout/activity_main.xml]: Not sure why you changed this? Since all our other fragments are dependent on this being unchanged

**Review by David-Huynh — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 4 above)
Comment (LiangbingQiangweiKaishaHexi1314) [reply app/build.gradle.kts:94]: yes

File: app/build.gradle.kts (hunk 3 above)
Comment (LiangbingQiangweiKaishaHexi1314) [reply app/build.gradle.kts]: These will remove.

File: app/build.gradle.kts (hunk 2 above)
Comment (LiangbingQiangweiKaishaHexi1314) [reply app/build.gradle.kts]: Added back.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (LiangbingQiangweiKaishaHexi1314) [reply app/src/main/res/values/strings.xml]: Removed

Comment (David-Huynh) [reply app/src/main/res/values/strings.xml]: it came back

**Review by David-Huynh — APPROVED** [review]

**Pull request closed without merging**